src/sheep_http.erl

```diff
@@ -35,18 +35,13 @@ upgrade(CowRequest0, Env, Handler, HandlerOpts) ->
     Bindings = cowboy_req:bindings(CowRequest0),
     Query = cowboy_req:parse_qs(CowRequest0),
     Peer = cowboy_req:peer(CowRequest0),
-    {Body, CowRequest} = case cowboy_req:has_body(CowRequest0) of
-               true ->
-                   {ok, Body0, CowRequest1} = cowboy_req:read_body(CowRequest0),
-                   {Body0, CowRequest1};
-               false ->
-                   {<<>>, CowRequest0}
-           end,
+    {Body, CowRequest} = read_body(CowRequest0),
+
     Request = #sheep_request{
         method = Method,
-        headers = Headers,
-        bindings = Bindings,
-        query = to_map(Query),
+        headers = Headers,                               %{binary() => binary()}
+        bindings = plist_to_map(maps:to_list(Bindings)), %{binary() => binary()}
```

Comment:
Bindings is atom to binary map, isn't it?
What is the purpose of doing "map to list" then "list to map"?

Reply:
Well, ok. I saw plist_to_map implementation and go it

Reply:
To stay backward compatible with existing code and to not introduce inconsistencies.